src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -232,5 +232,138 @@ public void RecordProperties_08()
                 Diagnostic(ErrorCode.ERR_FieldInitRefNonstatic, "O1").WithArguments("C1.O1").WithLocation(5, 33)
             );
         }
+
+        [Fact]
+        public void RecordEquals_01()
+        {
+            CompileAndVerify(@"
+using System;
+data class C(int X, int Y)
+{
+    public static void Main()
+    {
+        object c = new C(0, 0);
+        Console.WriteLine(c.Equals(c));
```

Comment:
>c.Equals(c) [](start = 26, length = 11)

I don't see how this doesn't call the method that throws null, or even why this code compiles with the conflict between the user-declared methods and the generated ones.  What am I missing?